solc/CommandLineInterface.cpp

```diff
@@ -916,67 +918,134 @@ void CommandLineInterface::serveLSP()
 		solThrow(CommandLineExecutionError, "LSP terminated abnormally.");
 }
 
-void CommandLineInterface::link()
+static bool isBytecodeChar(char c)
 {
-	solAssert(m_options.input.mode == InputMode::Linker, "");
+	return isxdigit(c) || c == '_' || c == '$';
```

Comment:
I think that the detection will unfortunately need to be a bit more complex to handle old-style placeholders since they could contain arbitrary chars. So it would be something like this on the outside of `__$` and `$__` and anything in between.

Maybe let's try this:
- The line has to contain `__`.
- All chars before the first and after the last `__` must be valid hex chars.
- There can be anything between the first `__` and last `__`.

Reply:
Unless we can make our lives easier and actually drop support for these old-style placeholders.
@chriseth Would that be acceptable without a breaking release?